dev_apps/music-nga/js/endpoint.js

```diff
@@ -292,7 +299,17 @@ function getArtist(filePath) {
     var artist = song.metadata.artist;
 
     return new Promise((resolve) => {
-      Database.enumerateAll('metadata.artist', artist, 'next', songs => resolve(songs));
+      Database.enumerateAll('metadata.artist', artist, 'next', (songs) => {
+        songs.sort((a, b) => {
+          var albumSort = a.metadata.album < b.metadata.album ?
+            -1 : (a.metadata.album > b.metadata.album ?  1 : 0);
+
+          return albumSort || (a.metadata.discnum - b.metadata.discnum) ||
+                              (a.metadata.tracknum - b.metadata.tracknum);
+        });
```

Comment:
We'd also probably want to sort the list of artists, but we could wait until later for that I suppose. I'm also not totally sure if sorting by album name is strictly necessary at the moment; if it is, we should probably be using `localeCompare`, since that should produce more useful results.